Add a note about TERRAFORM_CONFIG to help with migration old configs. (#34918)
hashicorp/terraform: website/docs/cli/config/environment-variables.mdx

export TF_CLI_CONFIG_FILE="$HOME/.terraformrc-custom"
```

> (Note that `TERRAFORM_CONFIG` is a deprecated alias for this variable. Please do not use `TERRAFORM_CONFIG` as it may be removed in future versions of Terraform.)

Reviewer: Small suggestion for active voice and style (in general we try to only use parenthesis in code):
```suggestion
Note that `TERRAFORM_CONFIG` is a deprecated alias for the `TF_CLI_CONFIG_FILE` variable. We recommend using `TF_CLI_CONFIG_FILE` instead of the deprecated `TERRAFORM_CONFIG` variable.
```
